src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -217,6 +224,35 @@ public void start(final Map<String, String> props) {
     return null;
   }
 
+  private final Optional<String> convertToJson(Optional<BsonDocument> bsonDocument, JsonWriterSettings jsonOptions) {
+    if (jsonOptions == null) {
+      return bsonDocument.map(BsonDocument::toJson);
+    }
+
+    return bsonDocument.map((doc) -> doc.toJson(jsonOptions));
+  }
+
+  private JsonWriterSettings handleJsonOptions(){
+    JsonMode jsonMode;
+    switch (sourceConfig.getJsonType()) {
```

Comment:
Second hint: you already have the JSON mode in a variable: `sourceConfig.getJsonType()`, there is no need to determine it again.